scripts/tests/provisioning/test_grub_gateway_migration.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+#
+# This script should be run to verify the functionality of the
+# dist-nilrt-grub-gateway IPK.
+# This test:
+#  * boots an already-provisioned NILRT QEMU virtual machine, using the RAUC
+#    (efi-ab) bootflow.
+#  * configures opkg with a locally-hosted ipk feed
+#  * installs and run the dist-nilrt-grub-gateway IPK, to reprovision the VM as
+#    a legacy safemode (grub) image
+#  * reconfigures opkg with the locally-build dist/ feed
+#  * installs and runs the grub-gateway ptest suite
+#
+#  This script returns 0 when the grub-gateway ptest suite passes, or a
+#  positive value otherwise.
+set -euo pipefail
+
+SCRIPT_ROOT=$(dirname "$BASH_SOURCE")
+
+usage () {
+	cat <<EOF
+$(basename $0) nilrt_vm_directory dist_feed_path other_ipk_feeds
+	       [other_ipk_feeds...]
+EOF
+	exit ${1:-2}
+}
+
+if [ $# -lt 3 ]; then
+	echo "ERROR: invalid or missing arguments." >&2
+	usage
+fi
+
+vm_dir=${1}
+dist_feed=${2}
+ipk_feeds=(${@:3})
+log_file=./$(basename ${BASH_SOURCE%.*}).log
+
+(
+	bash "${SCRIPT_ROOT}/start-vm-using-expect.sh" \
+		${ipk_feeds[@]/#/--ipk-feed /} \
+		--ipk-feed "${dist_feed}" \
+		"${vm_dir}" \
+		migrate_vm_to_safemode.expect
+
+	bash "${SCRIPT_ROOT}/start-vm-using-expect.sh" \
+		--ipk-feed "${dist_feed}" \
+		"${vm_dir}" \
+		test_safemode_provision.expect
+) | tee $log_file
```

Comment:
So are there going to be two log files for each test now? start-vm-using-expect.sh also [passes](https://github.com/ni/nilrt/blob/nilrt/master/dunfell/scripts/tests/provisioning/start-vm-using-expect.sh#L145) in a log file name to the expect files. I expected an accompanying change to stop doing that, but maybe I've misunderstood the big picture.

Reply:
That's correct. These new wrapper scripts aren't using that functionality anymore. I didn't bother removing it from the `start-` script because it isn't hurting anything and could potentially be useful to someone using that script directly.

Reply:
Can you share what the log file names all look like if you have a run on your system?

Reply:
For the migration test:
```
migrate_vm_to_safemode.log  # created by start- script; contains the VM session stdout from the tcl expect script of the same name
test_safemode_provision.log  # created by start- script; contains the VM session stdout from the tcl expect script of the same name
test_grub_gateway_migration.log  # created by the wrapper script; contains a cat of the above two, plus the setup/teardown done in the start- script
  # contains only the VM stdout from the tcl expect session
```

The test_provisioning wrapper creates similar - but there is only one VM session log.

Reply:
Thanks. IMO the multiple logs and their names are confusing. But we can fix it later if someone complains enough!